src/highlighters.cc

```diff
@@ -594,26 +594,58 @@ void show_whitespaces(const Context& context, HighlightFlags flags, DisplayBuffe
     }
 }
 
+template<bool relative>
 void show_line_numbers(const Context& context, HighlightFlags flags, DisplayBuffer& display_buffer, BufferRange)
 {
+    const Face face = get_face("LineNumbers");
+    const Face face_absolute = get_face("LineNumberAbsolute");
     LineCount last_line = context.buffer().line_count();
     int digit_count = 0;
     for (LineCount c = last_line; c > 0; c /= 10)
         ++digit_count;
 
     char format[] = "%?d│";
-    format[1] = '0' + digit_count;
-    const Face face = get_face("LineNumbers");
+    (relative) ? format[1] = '0' + digit_count + 1 : format[1] = '0' + digit_count;
+    int main_selection = (int)context.selections().main().cursor().line + 1;
     for (auto& line : display_buffer.lines())
     {
+        int current_line = (int)line.range().first.line + 1;
         char buffer[16];
-        snprintf(buffer, 16, format, (int)line.range().first.line + 1);
-        DisplayAtom atom{buffer};
-        atom.face = face;
-        line.insert(line.begin(), std::move(atom));
+        if (relative)
+        {
+            if (main_selection == current_line)
+            {
+                snprintf(buffer, 16, format, current_line);
+                DisplayAtom atom{buffer};
+                atom.face = face_absolute;
+                line.insert(line.begin(), std::move(atom));
+            }
+            else {
+                snprintf(buffer, 16, format, current_line - main_selection);
+                DisplayAtom atom{buffer};
+                atom.face = face;
+                line.insert(line.begin(), std::move(atom));
+            }
+        }
+        else {
+            snprintf(buffer, 16, format, current_line);
+            DisplayAtom atom{buffer};
+            (main_selection == current_line) ? atom.face = face_absolute
```

Comment:
for consistency i have chosen to highlight the main selection for the default and the relative line numbers